x-pack/plugins/security_solution/server/lib/timeline/routes/notes/get_notes.ts

```diff
@@ -139,11 +145,39 @@ export const getNotesRoute = (router: SecuritySolutionPluginRouter) => {
           const filterKueryNodeArray = [filterAsKueryNode];
 
           // retrieve all the notes created by a specific user
-          const userFilter = queryParams?.userFilter;
-          if (userFilter) {
-            filterKueryNodeArray.push(
-              nodeBuilder.is(`${noteSavedObjectType}.attributes.createdBy`, userFilter)
-            );
+          // the createdByFilter value is the uuid of the user
+          const createdByFilter = queryParams?.createdByFilter; // now uuid
+          if (createdByFilter) {
+            // because the notes createdBy property can be either full_name, email or username
+            // see pickSaveNote (https://github.com/elastic/kibana/blob/main/x-pack/plugins/security_solution/server/lib/timeline/saved_object/notes/saved_object.ts#L302)
+            // which uses the getUserDisplayName (https://github.com/elastic/kibana/blob/main/packages/kbn-user-profile-components/src/user_profile.ts#L138)
+            const [_, { security }] = await startServices();
+            const users: UserProfile[] = await security.userProfiles.bulkGet({
+              uids: new Set([createdByFilter]),
+            });
+            // once we retrieve the user by the uuid we can search all the notes that have the createdBy property with full_name, email or username values
+            if (users && users.length > 0) {
```

Comment:
Are there plans to support multi-user flows?

Reply:
no, not that I'm aware of. I had thought about this but because right now the UI only allows a single user I decided to go the simple route...